src/calendar/day/index.tsx

```diff
@@ -74,42 +53,37 @@ export default class Day extends Component<DayProps> {
       }
     }
     return label;
-  }
+  }, [marking]);
 
-  getAccessibilityLabel = memoize((day, marking, isToday) => {
+  const getAccessibilityLabel = useMemo(() => {
     const today = getDefaultLocale().today || 'today';
     const formatAccessibilityLabel = getDefaultLocale().formatAccessibilityLabel || 'dddd d MMMM yyyy';
-    const markingLabel = this.getMarkingLabel(marking);
 
-    return `${isToday ? today : ''} ${day.toString(formatAccessibilityLabel)} ${markingLabel}`;
-  });
-
-  getDayComponent() {
-    const {dayComponent, markingType} = this.props;
-
-    if (dayComponent) {
-      return dayComponent;
-    }
-    return markingType === 'period' ? PeriodDay : BasicDay;
-  }
-
-  render() {
-    const {day, marking} = this.props;
-    const date = day && xdateToData(day);
-    const _isToday = day ? isToday(day) : undefined;
-    const Component = this.getDayComponent();
-    const dayProps = omit(this.props, 'day');
-    const accessibilityLabel = this.getAccessibilityLabel(day, marking, _isToday);
-
-    return (
-      <Component
-        {...dayProps}
-        date={date}
-        testID={`${SELECT_DATE_SLOT}-${date?.dateString}`}
-        accessibilityLabel={accessibilityLabel}
-      >
-        {formatNumbers(day?.getDate())}
-      </Component>
-    );
-  }
-}
+    return `${_isToday ? today : ''} ${_day?.toString(formatAccessibilityLabel)} ${markingLabel}`;
+  }, [_day, marking, _isToday]);
+  
+  const dayProps = useMemo(() => {
```

Comment:
Why the dayProps has a dependency on the _day object? 
Also, consider destructing props you don't want to pass on at the top of the component.
Something like this
```
const {day, ...others} = props;
```
everything in `others` are stuff you want to pass forward. 
And if you want to use props from `others` internally, just use `others.marking` 

Does that make sense? I think it'll make the code cleaner and we can avoid redundant memoization. 


Actually the `useMemo` here is unnecessary anyway cause you don't create a new ref of anything you just extract the relevant props

Reply:
The `omit` creates a new object so I'm returning a new `dayProps` object every time. In this case, I need to pass all the props except `day`, so I don't see how using `...others` will assist here. Anyways, I'm removing the 'day' prop in favor of 'date' so I can now pass the props directly.